pkg/workers/sharings/share_data.go

```diff
@@ -529,7 +543,7 @@ func generateDirOrFilePatch(dirDoc *vfs.DirDoc, fileDoc *vfs.FileDoc) (*jsonapi.
 
 // getDocAtRecipient returns the revision of the document at the given
 // recipient.
-func getDocRevAtRecipient(doctype, docID string, recInfo *RecipientInfo) (string, error) {
+func getDocRevAtRecipient(newDoc *couchdb.JSONDoc, method string, doctype, docID string, recInfo *RecipientInfo) (string, error) {
```

Comment:
Right, I did a small refactoring to return the whole doc from this function and check the changes outside.